pkgs/applications/window-managers/qtile/default.nix

```diff
@@ -1,66 +1,127 @@
-{ lib, fetchFromGitHub, python37Packages, glib, cairo, pango, pkg-config, libxcb, xcbutilcursor }:
+{ lib
+, fetchFromGitHub
+, buildPythonApplication
+, wrapGAppsHook
+, cairo
+, cairocffi
+, dateutil
+, dbus-python
+, glib
+, gobject-introspection
+, libnotify
+, libpulseaudio
+, libxcb
+, mpd2
+, pango
+, pkg-config
+, psutil
+, pygobject3
+, pyxdg
+, setuptools
+, setuptools_scm
+, xcbutilcursor
+, xcffib
 
-let cairocffi-xcffib = python37Packages.cairocffi.override {
+, dbus
+, pytest
+, xcalc
+, xclock
+, xeyes
+, xorgserver
+, xterm
+, xvfb-run
```

Comment:
You also need librsvg and lm_sensors. Fontconfig may be needed but I have not yet fixed any of the fontconfig issues.